src/pybind/mgr/cephadm/serve.py

```diff
@@ -443,6 +446,32 @@ def _apply_all_services(self) -> bool:
 
         return r
 
+    def _apply_service_config(self, spec: ServiceSpec) -> None:
+        if spec.config:
+            section = utils.name_to_config_section(spec.service_name())
+            for k, v in spec.config.items():
+                k = k.replace(' ', '_')
+                try:
+                    current = self.mgr.get_foreign_ceph_option(section, k)
+                except KeyError:
+                    self.log.warning(
+                        f'Ignoring invalid {spec.service_name()} config option {k}'
+                    )
+                    continue
```

Comment:
raise and make it a HEALTH warning? I don't like silent fails, as they make it super hard to debug stuff. Like "why is this config option not working at all?" 

Also, would be great to validate them in the cli handler synchronously

Reply:
very tedious to raise a health warning from here, but i added validation at apply time, so this would only happen if a spec had a config option that was removed from ceph (and i think that's pretty harmless)

Reply:
might be harmless for the cluster, but not for users. Can you add something like 

```python
self.mgr.events.for_service(spec, 'ERROR', msg)
```

this is then displayed in https://docs.ceph.com/en/latest/cephadm/troubleshooting/#per-service-and-per-daemon-events  and https://tracker.ceph.com/issues/49262